instrumentation/kafka-clients/kafka-clients-common/library/src/main/java/io/opentelemetry/instrumentation/kafka/KafkaSingletons.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafka;
+
+import static io.opentelemetry.instrumentation.kafka.KafkaUtils.buildConsumerProcessInstrumenter;
+import static io.opentelemetry.instrumentation.kafka.KafkaUtils.buildConsumerReceiveInstrumenter;
+import static io.opentelemetry.instrumentation.kafka.KafkaUtils.buildProducerInstrumenter;
+
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+public final class KafkaSingletons {
```

Comment:
We only use those `...Singletons` classes in javaagent instrumentations - library instrumentations must allow constructing `Instrumenter`s from scratch with an `OpenTelemetry` instance passed by user (the `GlobalOpenTelemetry` is widely used by the agent, but in manually instrumented application we don't really recommend using it - we're recommending using dependency injection instead of a global var).

Reply:
Yeah ... this was left to decide.
I added KafkaUtils for that purpose -- so you don't bake-in instrumentation name.
I'll try to think of a better approach, e.g. adding more freedom to KafkaUtils.

I used KafkaSingletons so I can use it from both -- kafka-clients-0.11 javaagent and library.

Reply:
> I added KafkaUtils for that purpose -- so you don't bake-in instrumentation name.

You shouldn't bake-in the `OpenTelemetry` instance, or additional `AttributesExtractor`s, too - in general we usually have the whole Instrumenter-building logic inside the `...TracingBuilder` class and we just create a `...Tracing` instance in the javaagent instrumentation using it.

Reply:
Well, KafkaSingletons (which is now in my case KafkaUtils) already creates Instrumenters that my code can use, so I just reused that.

Reply:
But they're not extendable at all, and they use the global `OpenTelemetry` instance - we're trying to discourage the library instrumentation users from using the global. That's why in almost all other instrumentations in this repo `*Singletons` are placed in the javaagent (and they use the global) and use the `*TracingBuilder` to build instrumenters. The code will still be shared, you just move the bulk of the instrumenter-building code to the `*TracingBuilder` class.